src/centrality.rs

```diff
@@ -0,0 +1,298 @@
+        let is: usize = graph.to_index(node_s);
+        betweenness[is] = Some(0.0);
+    }
+    for node_s in graph.node_identifiers() {
+        let is: usize = graph.to_index(node_s);
+        let mut shortest_path_calc =
+            shortest_path_for_centrality(&graph, &node_s);
+        if endpoints {
+            _accumulate_endpoints(
+                &mut betweenness,
+                max_index,
+                &mut shortest_path_calc,
+                is,
+            );
+        } else {
+            _accumulate_basic(
+                &mut betweenness,
+                max_index,
+                &mut shortest_path_calc,
+                is,
+            );
+        }
+    }
+    _rescale(
+        &mut betweenness,
+        graph.node_count(),
+        normalized,
+        graph.is_directed(),
+        endpoints,
+    );
+
+    betweenness
+}
+
+fn _rescale(
+    betweenness: &mut Vec<Option<f64>>,
+    node_count: usize,
+    normalized: bool,
+    directed: bool,
+    endpoints: bool,
+) {
+    let mut do_scale = true;
+    let mut scale = 1.0;
+    if normalized {
+        if endpoints {
+            if node_count < 2 {
+                do_scale = false;
+            } else {
+                scale = 1.0 / (node_count * (node_count - 1)) as f64;
+            }
+        } else if node_count <= 2 {
+            do_scale = false;
+        } else {
+            scale = 1.0 / ((node_count - 1) * (node_count - 2)) as f64;
+        }
+    } else if !directed {
+        scale = 0.5;
+    } else {
+        do_scale = false;
+    }
+    if do_scale {
+        for x in betweenness.iter_mut() {
+            *x = x.map(|y| y * scale);
+        }
+    }
+}
+
+fn _accumulate_basic(
+    betweenness: &mut Vec<Option<f64>>,
+    max_index: usize,
+    path_calc: &mut ShortestPathData,
+    is: usize,
+) {
+    let mut delta = vec![0.0; max_index];
+    for w in &path_calc.verts_sorted_by_distance {
+        let iw = w.index();
+        let coeff = (1.0 + delta[iw]) / (path_calc.sigma[&iw] as f64);
+        let p_w = path_calc.predecessors.get_mut(w).unwrap();
+        for v in p_w {
+            let iv = (*v).index();
+            delta[iv] += (path_calc.sigma[&iv] as f64) * coeff;
+        }
+        if iw != is {
+            betweenness[iw] = betweenness[iw].map(|x| x + delta[iw]);
+        }
+    }
+}
+
+fn _accumulate_endpoints(
+    betweenness: &mut Vec<Option<f64>>,
+    max_index: usize,
+    path_calc: &mut ShortestPathData,
+    is: usize,
+) {
+    if betweenness[is].is_some() {
+        betweenness[is] = Some(
+            betweenness[is].unwrap()
+                + ((path_calc.verts_sorted_by_distance.len() - 1) as f64),
+        );
+    }
+    let mut delta = vec![0.0; max_index];
+    for w in &path_calc.verts_sorted_by_distance {
+        let iw = w.index();
+        let coeff = (1.0 + delta[iw]) / (path_calc.sigma[&iw] as f64);
+        let p_w = path_calc.predecessors.get_mut(w).unwrap();
```

Comment:
```suggestion
        let p_w = path_calc.predecessors.get(w).unwrap();
```

Reply:
Hmm. Don't know why I did this. I'm surprised the linting tools didn't complain about this.